docs/Testing.md

```diff
@@ -10,12 +10,153 @@ This isn't a hard and fast rule, but it does express our approach fairly well.
 
 ## Detox Testing
 
-MAVERICKTODO: Add info about detox here, or even a separate Testing-Detox.md.
+We provide a [README.md]([../boilerplate/detox/README.md]) inside of Ignite for how to get started with and run Detox tests once you have Ignited your app.
 
 ## Unit Testing
 
-MAVERICKTODO: Add info about unit testing with Jest here.
+> Unit tests cover the smallest parts of code, like individual functions or classes.
+
+-- <cite>[React Native docs](https://reactnative.dev/docs/testing-overview#unit-tests)</cite>
+
+### Example
+
+Ignite uses Jest as our test runner. Jest tests are written using `it` or `test` statements, which take a describe of the test, and a function to execute the test code.
+
+Then, inside the test function, we can make "assertions", or what we expect a value to be using the `expect` function. We pass the value as the first argument to the `expect` function, then we use one of the "matcher" methods on `expect`, such as `.toBe` to describe what the value should match.
+
+```ts
+it("given a date in the past, colorForDueDate() returns red", () => {
+  const input = colorForDueDate("2000-10-20")
+  expect(input).toBe("red")
+})
+```
+
+Jest functions like `it`, `test`, `expect`, and more are loaded globally by the Jest test runner, so you don't need to import them.
+
+### Usage
+
+In Ignite, we include unit tests for pure functions, such as models or utility functions.
+
+When writing a test, do your best to make sure that your tests include the following information:
+
+- Given - some precondition
+- When - some action executed by the function that you’re testing
+- Then - the expected outcome
+
+This is also known as AAA (Arrange, Act, Assert).
+
+You can read more about how to best practices for creating tests in the [Structuring Tests](https://reactnative.dev/docs/testing-overview#structuring-tests) section of the React Native docs.
+
+### Writing Tests
+
+To write your own tests, create a `.test.ts` file within the `app` or `test` directory.
+
+Then run `yarn test` to run all unit tests using Jest.
+
+When writing tests, you can also run Jest in watch mode by running `yarn test:watch`. This will start a long running Jest process, that re-runs your tests on save in your editor. This is useful for iterating on values and getting quick feedback about whether your changes were successful or not.
+
+### When to write Unit Tests
+
+Some developers believe that every line of code should have a co-responding test. However, Infinite Red believes that some unit tests provide more values than others. Here are a few situations where we have found unit tests to be helpful in client projects:
+
+- _complicated regex_: writing a series of valid and invalid inputs can help track down sneaky bugs
```

Comment:
Totally optional because everything reads great, but these 3 cases have a mix of capitalization and punctuation. Can make them consistent if you want.

Reply:
I'm pretty sure I got it consistent now, but if you could take another look, I would appreciate it